### ISSUE TITLE
Proposal using docker machine to set up registry 

### ISSUE THREAD

**Issue by agup006** [docker/machine#1663]
I love using machine to set up docker swarm, and thought the ease of use there could be translated to setting up a docker registry as well. There might need to be additional options when setting up the registry dedicated to the storage driver, but my thought is that this implementation would mimic the current swarm provisioning implementation.

**Comment by ehazlett** [comment]
We have discussed a "post-run" docker compose run feature.  I think instead of putting specific provisioning and creation things like this inside the core, it would be perfect for a post compose run.  Thoughts?

/cc @nathanleclaire

**Comment by agup006** [comment]
Yea, you are right. Having to add a hard coded set of options for specific containers becomes tedious. A post-run compose feature would be fantastic -- also in some cases allow people to deploy full scale applications without requiring docker on the deployer.

**Comment by nathanleclaire** [comment]
Yes, I am still very much in favor of running a compose file post-provision (and actually we might be able to move our Swarm management to this method), I wonder if it we could use the new [libcompose](https://github.com/docker/libcompose) for this.
